### PR TITLE
Make Collection::every() return true for empty sets.

### PR DESCRIPTION
**Pull request by markstory**
An empty list should pass all conditions as it is the vacuous truth. We incorrectly made Collection::every() return false in #8670. Returning true makes Collection::every() match implementations in most other languages/libraries.

Add additional tests to other filter/member checks to verify empty set behavior there as well.

Refs #10002

### CODE REVIEW

**Comment by ADmad** [comment]
:+1:

File: src/Collection/CollectionTrait.php [hunk 1]
```diff
@@ -89,9 +89,8 @@ public function reject(callable $c)
      */
     public function every(callable $c)
     {
-        $return = false;
+        $return = true;
```
Comment (nrother) [inline src/Collection/CollectionTrait.php]: The variable `$return` could be eliminated altogether, it always has the value of `false`.
